src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -63,7 +65,7 @@ export function getOccurrencesAcrossMultipleModels(registry: LanguageFeatureRegi
 	return first<ResourceMap<DocumentHighlight[]> | null | undefined>(orderedByScore.map(provider => () => {
 		return Promise.resolve(provider.provideMultiDocumentHighlights(model, position, otherModels, token))
```

Comment:
I believe this would be the place to filter down `otherModel`. I would do that my exposing the selector on the provider itself and then use `languageSelector.ts#score`